tweeter/logic.js

```diff
@@ -0,0 +1,124 @@
+function Tweeter() { 
+
+
+    let posts = [];
+
+
+    let postIdCounter = posts.length;
+
+
+    const getPosts = () => posts;
+
+
+    const addPost = function(text) {
+
+        if (text === '') {
+            return;
+        }
+
+        postIdCounter += 1;
+
+        const newId = `p${postIdCounter}`;
+        const newComments = [];
+
+        posts.push({
+            id: newId,
+            text: text,
+            comments: newComments
+        });
+    }
+
+    
+    const removePost = function(postId) {
+        const postNum = parseInt(postId.substring(1));
+        const numOfComments = posts[postNum - 1].comments.length;  
+
+        posts.splice(postNum - 1, 1);
+        
+        for (let i = postNum - 1; i < posts.length; i++) {
+            posts[i].id = `p${i + 1}`;
+        }
+        
+        postIdCounter -= 1;
+
+        for (let i = postNum - 1; i < posts.length; i++) {
+            for (let j = 0; j < posts[i].comments.length; j++) {
+                let commentNum = parseInt(posts[i].comments[j].id.substring(1));
+                commentNum -= numOfComments;
+                posts[i].comments[j].id = `c${commentNum}`
+            }
```

Comment:
When removing a post or comment you should not update the id. An ID should be unique. when updating all the ids you might run into the same ids